packages/valory/connections/test_libp2p/tests/test_dht.py

```diff
@@ -232,3 +244,135 @@ class TestCaseConfig:
         test_cls.__name__ = name
         test_cls.nodes = test_case.nodes
         globals()[test_cls.__name__] = test_cls
+
+
+RANGE_32_BIT = -1 << 31, (1 << 31) - 1
+
+base_message_strategy = dict(
+    dialogue_reference=st.tuples(st.text(), st.text()),
+    message_id=st.integers(*RANGE_32_BIT),
+    target=st.integers(*RANGE_32_BIT),
+)
+default_message_strategy = dict(
+    **base_message_strategy,
+    content=st.binary(),
+    performative=st.just(DefaultMessage.Performative.BYTES),
+)
+tendermint_message_strategy = st.one_of(
+    [
+        st.fixed_dictionaries(
+            dict(
+                **base_message_strategy,
+                performative=st.just(TendermintMessage.Performative.REQUEST),
+            )
+        ),
+        st.fixed_dictionaries(
+            dict(
+                **base_message_strategy,
+                performative=st.just(TendermintMessage.Performative.RESPONSE),
+                info=st.text(),
+            )
+        ),
+        st.fixed_dictionaries(
+            dict(
+                **base_message_strategy,
+                performative=st.just(TendermintMessage.Performative.ERROR),
+                error_code=st.sampled_from(CustomErrorCode),
+                error_msg=st.text(),
+                error_data=st.just({}),
+            )
+        ),
+    ],
+)
+
+
+class TestDHTRobustness(BaseP2PLibp2pTest, ACNWithBootstrappedEntryNodes):
+    """Test DHT Robustness"""
+
+    nodes = local_nodes
+    pytestmark = skip_if_ci_marker
+
+    def setup(self):
+        """Setup"""
+
+        for node in self.nodes:
+            self.make_connection(relay=False, entry_peers=[node.maddr])
+            self.make_client_connection(uri=node.uri, peer_public_key=node.public_key)
+
+    def teardown(self):
+        """Teardown"""
+
+        self._disconnect()
+        self.multiplexers.clear()
+        self.log_files.clear()
+
+    @pytest.mark.parametrize("exponent", [2, 3, 4])
+    def test_prolonged_message_exchange(self, exponent: int):
+        """Test prolonged message exchange"""
+
+        n_messages = 10**exponent
+        for _ in range(n_messages):
+            mux_pair = random.sample(self.multiplexers, 2)
+            sender, to = (c.address for m in mux_pair for c in m.connections)
+            envelope = self.enveloped_default_message(to=to, sender=sender)
+            mux_pair[0].put(envelope)
+            delivered_envelope = mux_pair[1].get(block=True, timeout=5)
+            assert self.sent_is_delivered_envelope(envelope, delivered_envelope)
+
+    @pytest.mark.parametrize("exponent", [2, 3, 4])
+    def test_ship_first_check_later(self, exponent: int):
```

Comment:
turning this up leads to issues, and more frequently so on public node setup. It's not just latency, at some point it fails to retrieve all messages that were put in the queue.